msal/src/main/java/com/microsoft/identity/client/TokenParameters.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param scopes
+     */
+    public void setScopes(List<String> scopes) {
+        this.mScopes = scopes;
+    }
+
+    /**
+     * Optional. If provided, will be used to force the session continuation.  If user tries to sign in with a different account, error
+     * will be returned.
+     *
+     * @return
+     */
+    public IAccount getAccount() {
+        return mAccount;
+    }
+
+    /**
+     * Optional. If provided, will be used to force the session continuation.  If user tries to sign in with a different account, error
+     * will be returned.
+     *
+     * @param account
+     */
+    public void setAccount(IAccount account) {
+        this.mAccount = account;
+    }
+
+    /**
+     * Optional. Can be passed to override the default authority.
+     * @return
+     */
+    public String getAuthority() {
+        return mAuthority;
+    }
+
+    /**
+     * Optional. Can be passed to override the default authority.
+     * @param authority
+     */
+    public void setAuthority(String authority) {
+        this.mAuthority = authority;
+    }
+
+    /**
+     * Optional. Can be passed into request specific claims in the id_token and access_token
+     * @return
+     */
+    public ClaimsRequest getClaimsRequest() {
+        return mClaimsRequest;
+    }
+
+    /**
+     * Optional. Can be passed into request specific claims in the id_token and access_token
+     * @param claimsRequest
+     */
+    public void setClaimsRequest(ClaimsRequest claimsRequest) {
+        this.mClaimsRequest = claimsRequest;
+    }
+
+    /**
+     * The Non-null {@link AuthenticationCallback} to receive the result back.
+     * 1) If user cancels the flow by pressing the device back button, the result will be sent
+     * back via {@link AuthenticationCallback#onCancel()}.
+     * 2) If the sdk successfully receives the token back, result will be sent back via
+     * {@link AuthenticationCallback#onSuccess(AuthenticationResult)}
+     * 3) All the other errors will be sent back via
+     * {@link AuthenticationCallback#onError(com.microsoft.identity.client.exception.MsalException)}.
+     * @return
+     */
+    public AuthenticationCallback getCallback() {
+        return mCallback;
+    }
+
+    /**
+     * The Non-null {@link AuthenticationCallback} to receive the result back.
+     * 1) If user cancels the flow by pressing the device back button, the result will be sent
+     * back via {@link AuthenticationCallback#onCancel()}.
+     * 2) If the sdk successfully receives the token back, result will be sent back via
+     * {@link AuthenticationCallback#onSuccess(AuthenticationResult)}
+     * 3) All the other errors will be sent back via
+     * {@link AuthenticationCallback#onError(com.microsoft.identity.client.exception.MsalException)}.
+     * @param callback
+     */
+    public void setCallback(AuthenticationCallback callback) {
+        this.mCallback = callback;
+    }
+
+    void setAccountRecord(AccountRecord record){
```

Comment:
Nit : java doc missing. Also should setAccoutRecord be public?

Reply:
AccountRecord is internal - the external representation is the IAccount